tutor/src/screens/my-courses/dashboard/index.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useCallback } from 'react'
+import { Button } from 'react-bootstrap'
+import { map, filter } from 'lodash'
+import styled from 'styled-components'
+import moment from 'moment'
+import { colors } from 'theme'
+import { Icon } from 'shared';
+import Tabs from '../../../components/tabs'
+import CourseInformation from '../../../models/course/information'
+import { useCoursesByOfferingId, useNumberOfStudents } from '../../../store/courses'
+import { useAllOfferings } from '../../../store/offering'
+import CreateACourse from './create-course'
+import CoursePreview from './preview-course'
+import ViewCourse from './view-course'
+import Resource from './resource'
+
+import { Offering, Course } from '../../../store/types'
+
+export interface OfferingWithCourses extends Offering {
+    courses: Course[]
+}
+
+const StyledMyCoursesDashboard = styled.div`
+    background-color: white;
+    padding: 3.8rem 1.6rem;
+    .controls {
+        display: flex;
+        justify-content: flex-end;
+    }
+    .offering-container {
+        margin: 1.6rem 0 1.6rem 0;
+        background-color: ${colors.neutral.bright};
+        padding-top: 2.4rem;
+        padding: 2.4rem 3.2rem 6.4rem;
+        min-height: 40rem;
+        h2 {
+            font-weight: 700;
+            font-size: 2.4rem;
+            line-height: 3rem;
+            color: ${colors.neutral.std};
+            margin-bottom: 1rem;
+        }
+        .tutor-tabs {
+            .nav-tabs {
+                li a {
+                    padding: 1rem 3rem;
+                }
+            }
+        }
+        .course-cards {
+            --gap: 3.2rem;
+            display: inline-flex;
+            flex-wrap: wrap;
+            margin-left: calc(-1 * var(--gap));
+            width: calc(100% + var(--gap));
+            > div {
+                margin: var(--gap) 0 0 var(--gap);
+                width: 260px;
+                margin-top: 0;
+                > div {
+                    &:hover {
+                        box-shadow: 0 5px 15px rgb(0 0 0 / 30%);
+                    }
+                }
+            }
+            .no-courses-message {
+                font-weight: 700;
+                font-size: 1.8rem;
+                color: ${colors.neutral.darker};
+                margin: 7rem auto;
+            }
+        }
+    }
+`
+
+const isCourseCurrent = (course: Course) => moment().isBefore(course.ends_at)
+const isCoursePast = (course: Course) => moment().isAfter(course.ends_at)
+const sortCurrentCourses = (courses: Course[]) => courses.sort((a, b) => {
+    // no students courses put them at the end of the list
+    if (useNumberOfStudents(a) === 0) { return -1 }
+    if(moment(a.ends_at).isAfter(b.ends_at)) { return 1 }
+    return 0
+})
+const sortPastCourses = (courses: Course[]) => courses.sort((a, b) => {
+    if(moment(a.ends_at).isAfter(b.ends_at)) { return 1 }
+    return 0
+})  
+
```

Comment:
This doesn't handle the a is before b case, sort should return -1 then.  Do we need to sort the courses without students as well?  With this courses will appear unsorted at the end.

We could make the sortPastCourses be `sortByCourseEndsAt` then:
```ts
const sortByCourseEndsAt = (courses: Course[]) => courses.sort((a, b) => {
     if(moment(a.ends_at).isAfter(b.ends_at)) { return 1 }
     if(moment(a.ends_at).isBefore(b.ends_at)) { return -1 }
     return 0
 })

const sortCurrentCourses = (courses: Course[]) => courses.sort((a, b) => {
   if (useNumberOfStudents(a) == 0 && useNumberOfStudents(b) != 0) { return -1 }
   return sortByCourseEndsAt(a, b);
}
```

Reply:
yeah, somehow I missed it. Thanks.